radioactivedecay/nuclide.py

```diff
@@ -0,0 +1,651 @@
+    _parse_nuclide_label,
+    _parse_decay_mode_label,
+    _check_fig_ax,
+)
+from radioactivedecay.utils import parse_radionuclide
+from radioactivedecay.utils import parse_nuclide
+
+
+class Nuclide:
+    """
+    ``Nuclide`` instances serve as name and atomic number/mass parsing
+    objects for any nuclide or element.
+
+    Parameters
+    ----------
+    input : Any
+        Input value for instantiation. Can be nuclide string in name
+        format (with or without hyphen), or atomic number (Z).
+    A : int, optional
+        For atomic number as input, atomic mass can be optionally
+        added for full nuclide string construction. 
+
+    Properties
+    ----------
+    name : str
+        Nuclide string if A data is given, otherwise the element
+        symbol is returned.
+    Z : int
+        Atomic number.
+    A : int
+        Atomic mass, returns 0 for raw elements.
+
+    Examples
+    --------
+    >>> rd.Nuclide('K-40')
+    Nuclide: K-40
+    >>> rd.Nuclide('K40')
+    Nuclide: K-40
+    >>> rd.Nuclide('K')
+    Element: K
+    >>> rd.Nuclide(19, 40)
+    Nuclide: K-40
+    >>> rd.Nuclide(19)
+    Element: K
+
+    """
+    
+    def __init__(
+        self, input: Any,
+        A: int = 0,
+        meta_state: str = "",
+        data: DecayData = DEFAULTDATA
+        ) -> None:
+        
+        Z_dict = {
+            1: "H", 2: "He", 3: "Li", 4: "Be", 5: "B", 6: "C", 7: "N", 8: "O",
+            9: "F", 10: "Ne", 11: "Na", 12: "Mg", 13: "Al", 14: "Si", 15: "P",
+            16: "S", 17: "Cl", 18: "Ar", 19: "K", 20: "Ca", 21: "Sc", 22: "Ti",
+            23: "V", 24: "Cr", 25: "Mn", 26: "Fe", 27: "Co", 28: "Ni", 29: "Cu",
+            30: "Zn", 31: "Ga", 32: "Ge", 33: "As", 34: "Se", 35: "Br", 36: "Kr",
+            37: "Rb", 38: "Sr", 39: "Y", 40: "Zr", 41: "Nb", 42: "Mo", 43: "Tc",
+            44: "Ru", 45: "Rh", 46: "Pd", 47: "Ag", 48: "Cd", 49: "In", 50: "Sn",
+            51: "Sb", 52: "Te", 53: "I", 54: "Xe", 55: "Cs", 56: "Ba", 57: "La",
+            58: "Ce", 59: "Pr", 60: "Nd", 61: "Pm", 62: "Sm", 63: "Eu", 64: "Gd",
+            65: "Tb", 66: "Dy", 67: "Ho", 68: "Er", 69: "Tm", 70: "Yb", 71: "Lu",
+            72: "Hf", 73: "Ta", 74: "W", 75: "Re", 76: "Os", 77: "Ir", 78: "Pt",
+            79: "Au", 80: "Hg", 81: "Tl", 82: "Pb", 83: "Bi", 84: "Po", 85: "At",
+            86: "Rn", 87: "Fr", 88: "Ra", 89: "Ac", 90: "Th", 91: "Pa", 92: "U",
+            93: "Np", 94: "Pu", 95: "Am", 96: "Cm", 97: "Bk", 98: "Cf", 99: "Es",
+            100: "Fm", 101: "Md", 102: "No", 103: "Lr", 104: "Rf", 105: "Db",
+            106: "Sg", 107: "Bh", 108: "Hs", 109: "Mt", 110: "Ds", 111: "Rg",
+            112: "Cn", 113: "Nh", 114: "Fl", 115: "Mc", 116: "Lv", 117: "Ts",
+            118: "Og"
+        }
+        sym_dict = dict((v,k) for k,v in Z_dict.items())
+        
+        self.Z_dict = Z_dict
+        self.sym_dict = sym_dict
+        self.parse_name(input, A, meta_state)
+        self.data = data
```

Comment:
Just FYI I renamed data -> decay_data when I refactored a load of stuff recently. Was trying to improve areas where my code was previously weaker, i.e. descriptiveness of variable names